From: SeongJae Park <sj@kernel.org>
Date: Sat, 14 Jun 2025 14:05:23 -0700
Subject: [PATCH] mm/damon: introduce [in]active memory ratio damos quota goal
 metric

DAMOS_LRU_[DE]PRIORITIZE can well be auto-tuned aiming a specific
active:inactive memory size ratio.  Add two new DAMOS quota goal metric
for the purpose, namely DAMOS_QUOTA_[IN]ACTIVE_MEM_BP.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 517f48ffe782..c2fc6c8cbbe4 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -179,6 +179,8 @@ enum damos_action {
  * @DAMOS_QUOTA_SOME_MEM_PSI_US:	System level some memory PSI in us.
  * @DAMOS_QUOTA_NODE_MEM_USED_BP:	MemUsed ratio of a node.
  * @DAMOS_QUOTA_NODE_MEM_FREE_BP:	MemFree ratio of a node.
+ * @DAMOS_QUOTA_ACTIVE_MEM_BP:		Active to [in]active memory ratio.
+ * @DAMOS_QUOTA_INACTIVE_MEM_BP:	Inactive to [in]active memory ratio.
  * @NR_DAMOS_QUOTA_GOAL_METRICS:	Number of DAMOS quota goal metrics.
  *
  * Metrics equal to larger than @NR_DAMOS_QUOTA_GOAL_METRICS are unsupported.
@@ -188,6 +190,8 @@ enum damos_quota_goal_metric {
 	DAMOS_QUOTA_SOME_MEM_PSI_US,
 	DAMOS_QUOTA_NODE_MEM_USED_BP,
 	DAMOS_QUOTA_NODE_MEM_FREE_BP,
+	DAMOS_QUOTA_ACTIVE_MEM_BP,
+	DAMOS_QUOTA_INACTIVE_MEM_BP,
 	NR_DAMOS_QUOTA_GOAL_METRICS,
 };
 
-- 
2.39.5

